integration-tests/hibernate-reactive-panache/src/test/java/io/quarkus/it/panache/reactive/PanacheFunctionalityTest.java

```diff
@@ -234,46 +236,67 @@ public void testTestTransaction2(UniAsserter asserter) {
     }
 
     @DisabledOnIntegrationTest
-    @ReactiveTransactional
+    @RunOnVertxContext
     @Test
     @Order(200)
     public void testReactiveTransactional(UniAsserter asserter) {
-        asserter.assertNotNull(() -> Panache.currentTransaction());
-        asserter.assertEquals(() -> Person.count(), 0l);
-        asserter.assertNotNull(() -> new Person().persist());
-        asserter.assertEquals(() -> Person.count(), 1l);
+        asserter.assertEquals(() -> reactiveTransactional(), 1l);
     }
 
-    @DisabledOnIntegrationTest
     @ReactiveTransactional
+    Uni<Long> reactiveTransactional() {
+        return Panache.currentTransaction()
+                .invoke(tx -> assertNotNull(tx))
+                .chain(tx -> Person.count())
+                .invoke(count -> assertEquals(0l, count))
+                .call(() -> new Person().persist())
+                .chain(tx -> Person.count());
+    }
+
+    @DisabledOnIntegrationTest
+    @RunOnVertxContext
     @Test
     @Order(201)
     public void testReactiveTransactional2(UniAsserter asserter) {
-        asserter.assertNotNull(() -> Panache.currentTransaction());
-        // make sure the previous one was NOT rolled back
-        asserter.assertEquals(() -> Person.count(), 1l);
-        // now delete everything and cause a rollback
-        asserter.assertEquals(() -> Person.deleteAll(), 1l);
-        asserter.execute(() -> Panache.currentTransaction().invoke(tx -> tx.markForRollback()));
+        asserter.assertTrue(() -> reactiveTransactional2());
     }
 
-    @DisabledOnIntegrationTest
     @ReactiveTransactional
+    Uni<Boolean> reactiveTransactional2() {
+        return Panache.currentTransaction()
+                .invoke(tx -> assertNotNull(tx))
+                .chain(tx -> Person.count())
+                .invoke(count -> assertEquals(1l, count))
+                .chain(() -> Person.deleteAll())
+                .invoke(count -> assertEquals(1l, count))
+                .chain(() -> Panache.currentTransaction())
+                .invoke(tx -> tx.markForRollback())
+                .map(tx -> true);
+    }
+
+    @DisabledOnIntegrationTest
+    @RunOnVertxContext
     @Test
     @Order(202)
     public void testReactiveTransactional3(UniAsserter asserter) {
-        asserter.assertNotNull(() -> Panache.currentTransaction());
-        // make sure it was rolled back
-        asserter.assertEquals(() -> Person.count(), 1l);
-        // and clean up
-        asserter.assertEquals(() -> Person.deleteAll(), 1l);
+        asserter.assertEquals(() -> testReactiveTransactional3(), 1l);
+    }
+
+    @ReactiveTransactional
```

Comment:
So `@ReactiveTransactional` can no longer be used on test methods?

Reply:
It can only be used if the method returns `Uni`. However, the `TestReactiveTransaction` works as before.